src/fiatconnect/saga.ts

```diff
@@ -189,27 +189,35 @@ export function* handleSubmitFiatAccount({
             yield put(submitFiatAccountKycApproved())
             yield delay(500) // Allow user to admire green checkmark
             break
-          // Denied, Expired, and Pending all fall through to the default case.
           case FiatConnectKycStatus.KycDenied:
             navigate(Screens.KycDenied, {
               flow,
               quote,
               retryable: true, // TODO: Get this dynamically once IHL supports it
             })
+            yield delay(500) // to avoid a screen flash
+            yield put(submitFiatAccountCompleted())
+            return
           case FiatConnectKycStatus.KycExpired:
             navigate(Screens.KycExpired, {
               flow,
               quote,
             })
+            yield delay(500) // to avoid a screen flash
+            yield put(submitFiatAccountCompleted())
+            return
           case FiatConnectKycStatus.KycPending:
             navigate(Screens.KycPending, {
               flow,
               quote,
             })
-          default:
             yield delay(500) // to avoid a screen flash
             yield put(submitFiatAccountCompleted())
             return
+          default:
+            throw new Error(
+              `Unrecognized FiatConnect KYC status "${fiatConnectKycStatus}" while attempting to handle quote selection for provider ${quote.getProviderId()}`
```

Comment:
I think it's the best we can do, short of implementing some other novel solution AFAICT. This will eventually loop the user back around to the SelectProvider screen to try again; they can also contact support if they like.